nominal/core/channel.py

```diff
@@ -114,6 +117,80 @@ def _from_conjure_logicalseries_api(
             _clients=clients,
         )
 
+    def get_decimated(
+        self,
+        start: str | datetime | IntegralNanosecondsUTC,
+        end: str | datetime | IntegralNanosecondsUTC,
+        *,
+        buckets: int | None = None,
+        resolution: int | None = None,
+    ) -> pd.DataFrame:
+        """Retrieve the channel data as a pandas.Dataframe, decimated to the given buckets or resolution.
+
+        Enter either the number of buckets or the resolution for the output.
+        Resolution in picoseconds for picosecond-granularity dataset, nanoseconds otherwise.
+        """
+        if buckets is not None:
+            # Somehow the number of points returned is buckets / 1000
+            buckets = buckets * 1000
+
+        result = self._decimate_request(start, end, buckets, resolution)
+
+        def to_ts(timestamp: api_Timestamp) -> pd.Timestamp:
+            return pd.Timestamp(timestamp.seconds, unit="s", tz="UTC") + pd.Timedelta(timestamp.nanos, unit="ns")
```

Comment:
this doesn't create/utilize a closure so I'd just define this outside of this function